addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -368,6 +372,59 @@ const Wysiwyg = Widget.extend({
     redo: function () {
         this.odooEditor.historyRedo();
     },
+    /**
+     * Start or resume the date field changes muation observers.
+     *
+     * Necessary to keep all copies of a given field at the same value throughout the page.
+     */
+    _observeDateFieldChanges: function () {
+        if(this.dateObservers) {
+            for (let observerData of this.dateObservers) {
+                observerData.observer.observe(
+                    observerData.field,
+                    { characterData:true, subtree: true, childList: true }
+                );
+            }
+        } else {
+            const dateFieldSelector = '*[data-oe-field][data-oe-type=datetime], *[data-oe-field][data-oe-type=date]'
+            const $dateFields = this.$editable.find(dateFieldSelector);
+            this.dateObservers = [];
+
+            $dateFields.each((i, field) => {
+                const observer = new MutationObserver(() => {
+                        this._pauseDateFieldObservers();
+
+                        const $field = $(field);
+                        let selector = '[data-oe-id="' + $field.data('oe-id') + '"]';
+                        selector += '[data-oe-field="' + $field.data('oe-field') + '"]';
+                        selector += '[data-oe-model="' + $field.data('oe-model') + '"]';
+                        this.$editable.find(selector).not($field).html($field.html()).addClass('o_editable_date_field_format_changed');
+
+                        this._observeDateFieldChanges();
+                    }
+                );
+                observer.observe(field, {characterData: true, subtree: true, childList: true });
+                this.dateObservers.push({field: field, observer: observer});
+            });
+
+            this.$editable.on('mousedown',dateFieldSelector, function(e){
+                console.log('mousedown event ', e);
+                const $field = $(this);
+                if (!$field.hasClass('o_editable_date_field_format_changed')) {
+                    $field.html($field.data('oe-original-with-format'));
+                    $field.addClass('o_editable_date_field_format_changed');
+                }
```

Comment:
This code belongs to another method for me as it is not related to the observation.
Especially if the method will be more generic as mentioned in another comment.